### PR TITLE
added function to fetch for patient phone number

### PR DESCRIPTION
**Pull request by pamzey**
I have added a function that retrieves a patient's phone number, if available, which will be used when sending SMS reminders for the next appointment. If the phone number is not available, the function will not send the SMS and will instead display a message to the user indicating that no phone number is linked to the patient.

### CODE REVIEW

File: app/controllers/api/v1/send_sms_controller.rb [hunk 1]
```diff
@@ -1,62 +1,41 @@
...
-          render json: config['development']
-        else
-          render json: { error: 'Development configuration not found' }, status: :not_found
-        end
-
-      rescue Errno::ENOENT
-        render json: { error: 'Configuration file not found' }, status: :not_found
-      end
-
+    if config.key?('development')
```
Comment (fchiyenda) [inline app/controllers/api/v1/send_sms_controller.rb]: Was thinking we can consider other environments as well

File: app/controllers/api/v1/send_sms_controller.rb [hunk 2]
```diff
@@ -1,62 +1,41 @@
...
     config_file = Rails.root.join('config', 'application.yml')
     config = YAML.load_file(config_file)
 
     params.each do |key, value|
-
-      if config['development'].key?(key)
-         config['development'][key] = value
-      end
-      
+      config['development'][key] = value if config['development'].key?(key)
```
Comment (fchiyenda) [inline app/controllers/api/v1/send_sms_controller.rb]: Let us not tied it to a specific environment

File: app/controllers/api/v1/send_sms_controller.rb [hunk 3]
```diff
@@ -121,20 +95,36 @@ def service
...
+    ImmunizationService::SendSmsService.perform_async(date, details)
+  rescue => e
+    "Failed to queue SMS: #{e.message}"
   end
 
   def validatephone(phone)
-    phone_pattern = /\A\+\d{12}\z/
     phone = "+265" + phone[1..] if phone.present? && phone.starts_with?('0')
-    return phone if phone_pattern.match?(phone)
+    phone_pattern = /\A\+\d{12}\z/
```
Comment (fchiyenda) [inline app/controllers/api/v1/send_sms_controller.rb:120]: Should we retrict to Malawi numbers only?

File: config/routes.rb [hunk 4]
```diff
@@ -421,4 +421,5 @@ def paginate_url(url, params)
   post '/api/v1/send_sms', to: 'api/v1/send_sms#index'
   post '/api/v1/sms_configuration', to: 'api/v1/send_sms#update'
   get '/api/v1/configurations', to: 'api/v1/send_sms#show'
+  get '/api/v1/patient_phone', to: 'api/v1/send_sms#fetch_phone'
```
Comment (fchiyenda) [inline config/routes.rb:424]: Just thinking isn't the clients phone number already available in the frontend by the time we would like to send an sms or the time we are trying to show the send sms popup

File: app/controllers/api/v1/send_sms_controller.rb (hunk 2 above)
Comment (pamzey) [reply app/controllers/api/v1/send_sms_controller.rb]: I have fixed this what i have done it checks the environment being used and then it works with that

File: app/controllers/api/v1/send_sms_controller.rb (hunk 1 above)
Comment (pamzey) [reply app/controllers/api/v1/send_sms_controller.rb]: I have fixed this issue

File: app/controllers/api/v1/send_sms_controller.rb (hunk 2 above)
Comment (pamzey) [reply app/controllers/api/v1/send_sms_controller.rb]: I have added the fix which was overwriting application.yml file contents

Comment (pamzey) [reply app/controllers/api/v1/send_sms_controller.rb]: I have removed the unnecessary code for checking rails enviroment

**Review by fchiyenda — CHANGES_REQUESTED** [review]
As we discussed on the Validations

**Review by fchiyenda — APPROVED** [review]

**Pull request merged**